src/CommunityToolkit/Xamarin.CommunityToolkit/Views/ViewSwitcher/TextSwitcherRenderer.android.cs

```diff
@@ -0,0 +1,558 @@
+
+			if (e.OldElement != null)
+			{
+				e.OldElement.PropertyChanged -= OnElementPropertyChanged;
+				this.MaybeRequestLayout();
+			}
+
+			if (e.NewElement != null)
+			{
+				this.EnsureId();
+
+				if (visualElementTracker == null)
+				{
+					visualElementTracker = new VisualElementTracker(this);
+				}
+
+				e.NewElement.PropertyChanged += OnElementPropertyChanged;
+
+				UpdateText();
+				UpdateLineHeight();
+				UpdateCharacterSpacing();
+				UpdateTextDecorations();
+				UpdateTransition();
+				if (e.OldElement?.LineBreakMode != e.NewElement.LineBreakMode)
+					UpdateLineBreakMode();
+				if (e.OldElement?.HorizontalTextAlignment != e.NewElement.HorizontalTextAlignment || e.OldElement?.VerticalTextAlignment != e.NewElement.VerticalTextAlignment)
+					UpdateGravity();
+				if (e.OldElement?.MaxLines != e.NewElement.MaxLines)
+					UpdateMaxLines();
+
+				UpdatePadding();
+
+				ElevationHelper.SetElevation(this, e.NewElement);
+			}
+		}
+
+		protected virtual void OnElementPropertyChanged(object sender, PropertyChangedEventArgs e)
+		{
+			if (this.IsDisposed())
+			{
+				return;
+			}
+
+			ElementPropertyChanged?.Invoke(this, e);
+
+			if (Control?.LayoutParameters == null && hasLayoutOccurred)
+				return;
+
+			if (e.PropertyName == Label.HorizontalTextAlignmentProperty.PropertyName || e.PropertyName == Label.VerticalTextAlignmentProperty.PropertyName)
+				UpdateGravity();
+			else if (e.PropertyName == Label.TextColorProperty.PropertyName ||
+				e.PropertyName == Label.TextTypeProperty.PropertyName)
+				UpdateText();
+			else if (e.PropertyName == Label.FontProperty.PropertyName)
+				UpdateText();
+			else if (e.PropertyName == Label.LineBreakModeProperty.PropertyName)
+				UpdateLineBreakMode();
+			else if (e.PropertyName == Label.CharacterSpacingProperty.PropertyName)
+				UpdateCharacterSpacing();
+			else if (e.PropertyName == Label.TextDecorationsProperty.PropertyName)
+				UpdateTextDecorations();
+			else if (e.PropertyName == Label.TextProperty.PropertyName ||
+				e.PropertyName == Label.FormattedTextProperty.PropertyName ||
+				e.PropertyName == Label.TextTransformProperty.PropertyName)
+				UpdateText();
+			else if (e.PropertyName == Label.LineHeightProperty.PropertyName)
+				UpdateLineHeight();
+			else if (e.PropertyName == Label.MaxLinesProperty.PropertyName)
+				UpdateMaxLines();
+			else if (e.PropertyName == Label.PaddingProperty.PropertyName)
+				UpdatePadding();
+			else if (e.PropertyName == ViewSwitcher.TransitionDurationProperty.PropertyName ||
+				e.PropertyName == ViewSwitcher.TransitionTypeProperty.PropertyName)
+				UpdateTransition();
+		}
+
+		void UpdateColor()
+		{
+			var color = Element.TextColor;
+			if (color == lastUpdateColor)
+				return;
+			lastUpdateColor = color;
+
+			if (color.IsDefault)
+				children.ForEach(c => c.SetTextColor(labelTextColorDefault));
+			else
+				children.ForEach(c => c.SetTextColor(color.ToAndroid()));
```

Comment:
Let's make sure to just call the `ToAndroid()` just once
```suggestion
                                var androidColor = color.ToAndroid();
				children.ForEach(c => c.SetTextColor(androidColor ));
```